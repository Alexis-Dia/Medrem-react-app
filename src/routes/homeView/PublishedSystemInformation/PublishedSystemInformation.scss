.published-system-information {
  width: 930px;
  margin: 50px auto;
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);

  h3 {
    text-align: center;
    padding: 10px 0;
    border-bottom: 1px solid;
    margin-bottom: 0;
  }

  ul {
    list-style: none;
    overflow-y: auto;
    height: 300px;
    padding: 0;

    li {
      padding: 15px;

      &:hover {

        background: rgba(0, 0, 0, 0.2);
      }
    }
  }
}
