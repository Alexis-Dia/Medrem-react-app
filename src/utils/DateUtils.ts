class DateUtils {

    static readonly DATE_TIME_MASK = [
        /\d/,
        /\d/,
        /\d/,
        /\d/,
        "/",
        /\d/,
        /\d/,
        "/",
        /\d/,
        /\d/,
        " ",
        /\d/,
        /\d/,
        ":",
        /\d/,
        /\d/,
    ];

    static readonly DATE_TIME_FORMAT_DEFAULT = "YYYY/MM/DD HH:mm";
    static readonly DATE_YY_MM_DD_FORMAT = "yy/MM/dd";

    private static readonly VALID_PARTS_LENGTH = 4;

    static parseDate = (dateStr: string, format: string): Date | null => {
        const regex = format.toLocaleLowerCase()
            .replace(/\bd+\b/, "(?<day>\\d+)")
            .replace(/\bm+\b/, "(?<month>\\d+)")
            .replace(/\by+\b/, "(?<year>\\d+)");

        const parts = new RegExp(regex).exec(dateStr);
        if (parts) {
            //@ts-ignore
            const { year, month, day } = parts.groups;
            return parts.length === DateUtils.VALID_PARTS_LENGTH ? new Date(year, month - 1, day) : null;
        }
        return null;
    }
}

export default DateUtils;
