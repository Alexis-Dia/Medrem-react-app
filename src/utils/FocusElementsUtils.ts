class FocusElementsUtils {

    private static getComboBox = (element: any): HTMLElement | null => {

        for (const childNode of element.childNodes) {
            let attribute;
            if (childNode instanceof Element) {
                attribute = childNode.getAttribute("aria-haspopup");
            }
            if (attribute) {
                return childNode;
            } else {
                const comboBox = FocusElementsUtils.getComboBox(childNode);
                if (comboBox) {
                    return comboBox;
                }
            }
        }
        return null;
    }

    private static getInputOrTextarea = (element: any): HTMLInputElement | null => {

        const inputs = element.getElementsByTagName("input");
        if (inputs && inputs.length > 0) {
            return inputs[0];
        }
        const textareas = element.getElementsByTagName("textarea");
        if (textareas && textareas.length > 0) {
            for (const textarea of textareas) {
                if (textarea.getAttribute("aria-invalid")) {
                    return textarea;
                }
            }
            return textareas[textareas.length - 1];
        }
        return null;
    }

    private static getButton = (element: any): HTMLElement | null => {

        const buttons = element.getElementsByTagName("button");
        if (!buttons) {
            return null;
        }

        return buttons[0];
    }

}

export default FocusElementsUtils;
