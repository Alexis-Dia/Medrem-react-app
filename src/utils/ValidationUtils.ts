import { ValidationRule } from "../types/validation";

class ValidationUtils {

    static getRuleByFieldName = (fieldName: string): ValidationRule => {
        switch (fieldName) {
            case "email":
                return ValidationRule.EMAIL;
            case "password":
                return ValidationRule.PASSWORD;
            case "username":
                return ValidationRule.USERNAME;
            case "firstName":
            case "lastName":
                return ValidationRule.VALUE_REQUIRED;
            case "externalEmployeeId":
                return ValidationRule.EMPLOYEE;
            default:
                return ValidationRule.VALUE_REQUIRED;
        }
    };

}

export default ValidationUtils;
