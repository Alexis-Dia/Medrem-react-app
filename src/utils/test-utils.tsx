/* istanbul ignore file */

import React from "react";
import { IntlProvider } from "react-intl";
import { Provider } from "react-redux";
import { MuiThemeProvider } from "material-ui/styles";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

export enum Wrapper {
    REDUX_PROVIDER,
    INTL_PROVIDER,
    THEME_PROVIDER
}

export const getWrappedComponent = (component: JSX.Element, wrappers: Array<Wrapper>, messages = {}, store = {}): JSX.Element => {
    let wrappedComponent = component;
    wrappers.forEach((wrapper) => {
        wrappedComponent = wrapComponent(wrappedComponent, wrapper, messages, store);
    });
    return wrappedComponent;
};

const wrapComponent = (component: JSX.Element, wrapper: Wrapper, messages: any, store: any) => {
    switch (wrapper) {
        case Wrapper.INTL_PROVIDER:
            return injectIntl(component, messages);
        case Wrapper.REDUX_PROVIDER:
            return injectStore(component, store);
        case Wrapper.THEME_PROVIDER:
            return injectStyles(component);
        default:
            return component;
    }
};

const injectStore = (node: JSX.Element, store: any): JSX.Element => (
    <Provider store={store}>
        {node}
    </Provider>
);

const injectIntl = (node: JSX.Element, messages: any): JSX.Element => (
    <IntlProvider locale="en" defaultLocale="en" messages={messages}>
        {node}
    </IntlProvider>
);

const injectStyles = (node: JSX.Element): JSX.Element => (
    <MuiThemeProvider>
        {node}
    </MuiThemeProvider>
);

export const findByAriaLabel = (component: any, ariaLabel: string) => component.find(`[aria-label='${ariaLabel}']`);

interface MountingParams {
    component: JSX.Element;
    name: string;
}

export const mountComponent = (params: MountingParams) => {
    const wrapper = mount(params.component);
    return {
        wrapper,
        component: wrapper.find(params.name),
    };
};

export const renderComponent = (params: MountingParams) => renderer.create(params.component);
