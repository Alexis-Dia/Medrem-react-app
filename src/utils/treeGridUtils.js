export function getTreeArr(arr) {
    const newArr = [];
    let mainGeneratedId = 0;
    let generatedId = 100;
    arr.forEach((item) => {
        Array.isArray(newArr)&& newArr.push({
            "parentId": null,
            "id": mainGeneratedId,
            "name": item.name,
        });
        mainGeneratedId = mainGeneratedId + 1;
        let currentId = 0;
        if (Array.isArray(item.fields)&& item.fields.length > 0) {
            currentId = mainGeneratedId - 1;
        }
        Array.isArray(item.fields)&& item.fields.forEach((part) => {
            newArr.push({
                "parentId": currentId,
                "id": mainGeneratedId,
                "name": part.name,
                "parentName": item.name,
                "selectedByDefault": part.selectedByDefault,
            });
            mainGeneratedId = mainGeneratedId + 1;
            generatedId = generatedId + 1;
        });
    });
    generatedId = 0;
    return newArr;
}
