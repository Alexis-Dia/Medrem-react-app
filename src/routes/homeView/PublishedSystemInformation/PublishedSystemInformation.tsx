import React from "react";
import { connect, ConnectedProps } from "react-redux";
import { FormattedMessage } from "react-intl";
import "./PublishedSystemInformation.scss";
import { RootState } from "../../../store/reducers";

type Props = PropsFromRedux;

const PublishedSystemInformation = (props: Props) => {
    const { systemInformations, language } = props;

    return (
        systemInformations.length > 0 ?
            <div className="published-system-information">
                <h3>
                    <FormattedMessage id="systemInformation" defaultMessage="System Information" />
                </h3>
                <ul>
                    {systemInformations.map((systemInformation) => {
                        const foundTranslation = systemInformation.translations.filter((translation) => language && translation.language.toLowerCase() === language.toLowerCase())[0];
                        const text = foundTranslation ? foundTranslation.text : "";
                        return (
                            <li key={systemInformation.id}>
                                <div className="information-text">
                                    {`${systemInformation.informationType} | ${text}`}
                                </div>
                            </li>
                        );
                    })}
                </ul>
            </div> :
            <div />
    );
};

const mapStateToProps = (state: RootState) => ({
    language: state.locale.lang || state.auth.user.language,
    systemInformations: state.system.currentlyPublished,
});

const connector = connect(mapStateToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(PublishedSystemInformation);
