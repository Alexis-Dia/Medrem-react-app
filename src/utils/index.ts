import ValidationUtils from "./ValidationUtils";

export {
    ValidationUtils,
};
