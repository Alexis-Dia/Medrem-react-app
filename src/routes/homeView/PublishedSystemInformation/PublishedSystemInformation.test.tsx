import React from "react"
import PublishedSystemInformation from "./PublishedSystemInformation"
import configureStore from "redux-mock-store";
import { getWrappedComponent, mountComponent, renderComponent, Wrapper } from "../../../utils/test-utils";

const systemInformation = {
    id: 5,
    informationType: "update",
    translations: [{ language: "EN", text: "en-text" }]
};

const mockStore = configureStore();
const initialStore = mockStore({
    locale: { lang: "EN" },
    system: {
        currentlyPublished: [systemInformation],
    },
});

const getWrappedComponentImpl = (store = initialStore) => {
    const wrappers = [Wrapper.REDUX_PROVIDER, Wrapper.INTL_PROVIDER];
    const component = <PublishedSystemInformation />;
    return { component: getWrappedComponent(component, wrappers, {}, store), name: "PublishedSystemInformation" };
};

describe("<PublishedSystemInformation />", () => {
    const { wrapper: wrapperWithInitialProps, component: componentWithInitialProps } = mountComponent(getWrappedComponentImpl());

    afterEach(() => {
        
    });

    it("renders correctly", () => {
        const tree = renderComponent(getWrappedComponentImpl());
        expect(tree).toMatchSnapshot();
    });
    
    it("renders with initial props", () => {
        const { locale, system } = initialStore.getState();
        expect(componentWithInitialProps.props()).toMatchObject({
            systemInformations: system.currentlyPublished,
            language: locale.lang,
        });
    });

    it("should pass props to children", () => {
        const list = wrapperWithInitialProps.find("ul");
        const listItems = list.find("li");
        expect(listItems).toHaveLength(1);

        const firstItem = listItems.first();
        expect(firstItem.key()).toBe(String(systemInformation.id));
        expect(firstItem.find("div").text()).toBe("update | en-text");
    });

    it("renders without system informations", () => {
        const store = mockStore({
            ...initialStore.getState(),
            system: {
                currentlyPublished: [],
            },
        });
        const { wrapper } = mountComponent(getWrappedComponentImpl(store));
        expect(wrapper.find("div")).toHaveLength(1);
    });
});