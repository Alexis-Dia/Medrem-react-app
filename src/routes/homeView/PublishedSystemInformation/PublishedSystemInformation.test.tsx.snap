// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublishedSystemInformation /> renders correctly 1`] = `
<div
  className="published-system-information"
>
  <h3>
    System Information
  </h3>
  <ul>
    <li>
      <div
        className="information-text"
      >
        update | en-text
      </div>
    </li>
  </ul>
</div>
`;
